cmake_uninstall.cmake.in

```diff
@@ -18,16 +18,25 @@ IF(NOT EXISTS "@CMAKE_CURRENT_BINARY_DIR@/install_manifest.txt")
   RETURN()
 ENDIF(NOT EXISTS "@CMAKE_CURRENT_BINARY_DIR@/install_manifest.txt")
 
+IF(NOT DEFINED PACKAGE_CREATES_DOT_CATKIN OR NOT "${PACKAGE_PREVIOUS_INSTALL_PREFIX}" EQUAL "${CMAKE_INSTALL_PREFIX}")
+  SET(PACKAGE_PREVIOUS_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}" CACHE INTERNAL "Cache install prefix given to the package")
+  IF(EXISTS "${CMAKE_INSTALL_PREFIX}/.catkin")
+    SET(PACKAGE_CREATES_DOT_CATKIN FALSE CACHE INTERNAL "")
+  ELSE()
+    SET(PACKAGE_CREATES_DOT_CATKIN TRUE CACHE INTERNAL "")
+  ENDIF()
+ENDIF()
```

Comment:
Sorry for the delay, I have been busy lately but I think I figured the issue.

There's a few things to note:
- this script only runs after at least one succesfull install (we skip the run if the `install_manifest.txt` file does not exist)
- this script is only run via `cmake -P`, it cannot modify the cache (in fact I am surprised it can use it)

So I think the correct approach is to set `PACKAGE_CREATES_DOT_CATKIN` from the `_SETUP_PROJECT_UNINSTALL` macro and pass its value to the script, then `PACKAGE_CREATES_DOT_CATKIN` should be the right value to use.